test/handler.test.js

```diff
@@ -361,6 +361,130 @@ describe('strong-error-handler', function() {
+        .expect(/500(.*?)a test error message/)
+        .expect(/extra(.*?)sensitive data/)
+        .expect(/id="stacktrace"(.*?)ErrorWithProps: a test error message/,
+          done);
+    });
+
+    it('contains subset of properties when status=4xx', function(done) {
+      var error = new ErrorWithProps({
+        name: 'ValidationError',
+        message: 'The model instance is not valid.',
+        statusCode: 422,
+        details: 'some details',
+        extra: 'sensitive data',
+      });
+      givenErrorHandlerForError(error, {debug: false});
+      requestHTML()
+        .end(function(err, res) {
+          expect(res.statusCode).to.eql(422);
+          var body = res.error.text;
+          expect(body).to.not.match(/sensitive data/);
+          expect(body).to.match(/<title>ValidationError<\/title>/);
+          expect(body).to.match(/422(.*?)The model instance is not valid./);
+          //the stack trace html should be empty
+          expect(body).to.match(/<pre id="stacktrace"><\/pre>/);
+          done();
+        });
+    });
+
+    it('html contains only safe info when status=5xx', function(done) {
+      // Mock an error reported by fs.readFile
+      var error = new ErrorWithProps({
+        name: 'Error',
+        message: 'ENOENT: no such file or directory, open "/etc/passwd"',
+        errno: -2,
+        code: 'ENOENT',
+        syscall: 'open',
+        path: '/etc/password',
+      });
+      givenErrorHandlerForError(error);
+
+      requestHTML()
+        .end(function(err, res) {
+          expect(res.statusCode).to.eql(500);
+          var body = res.error.text;
+          expect(body).to.not.match(/\/etc\/password/);
+          expect(body).to.not.match(/-2/);
+          expect(body).to.not.match(/ENOENT/);
+          // only have the following
+          expect(body).to.match(/<title><\/title>/);
+          expect(body).to.match(/500(.*?)Internal Server Error/);
+          done();
+        });
+    });
+
+    function requestHTML(url) {
+      return request.get(url || '/')
+        .set('Accept', 'text/html')
+        .expect('Content-Type', /^text\/html/);
+    }
+  });
+
+  context('Content Negotiation', function() {
+    it('should honor default to json without options', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {});
+      request.get('/')
+        .set('Accept', '*/*')
+        .expect('Content-Type', /^application\/json/, done);
+    });
+
+    it('should honor accepted content-type', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {
+        defaultType: 'application/json',
+      });
+      request.get('/')
+        .set('Accept', 'text/html')
+        .expect('Content-Type', /^text\/html/, done);
+    });
+
+    it('considers order of Accepted content-types', function(done) {
+      givenErrorHandlerForError(new Error('Some error'), {});
+      request.get('/')
+        // `*/*` will be used because its provided first
```

Comment:
Interesting. IIUC, when `*/*` is provided first, we will pick the first supported type, which is JSON by our preference. Can we change this so that user-provided `defaultType` is picked instead?